Update automate.sh

GithubUser699 opened the pull request:
I use the output of $XDG_TEMPLATES_DIR to get the translated path and check if it already exists or not. If so, I'll just move the contents of the *Templates* folder; if not, I'll first create that folder and than do the former thing.
As you can see I deleted rsync as it isn't needed any longer.

GithubUser699 committed:
Update automate.sh

I use the output of $XDG_TEMPLATES_DIR to get the translated path and check if it already exists or not. If so, I'll just move the contents of the *Templates* folder; if not, I'll first create that folder and than do the former thing.
As you can see I deleted rsync as it isn't needed any longer.

GithubUser699 commented:
I just tested it with another, english system.
The (updated) script does work, no matter if I already have a `Templates` folder or if I don't and no matter if the `user-dirs.dirs` file is already there or not.
Script and output works as expected on both systems - one in english, one in german.

GithubUser699 committed:
Update automate.sh

removed a typo in a comment and added *-i*, so the user get's prompted if the file already exists + added a note accordingly in the "echo" print.

GithubUser699 committed:
Update automate.sh

updated typo in the comments...

angela-d commented on automate.sh:
```diff
@@ -1,45 +1,45 @@
+		else
+			# this config already exists; append the entry
+			source ~/.config/user-dirs.dirs
+			customPath=$XDG_TEMPLATES_DIR
```
Where is `$XDG_TEMPLATES_DIR` being set? 
See comment to the left, too. :)

GithubUser699 replied:
I am not sure if I understand your question.

If you run `xdg-user-dirs-update` the `~/.config/user-dirs.dirs` file is being created. It contains a few paths for your system, which can be called using `$XDG_TEMPLATES_DIR` after using source to let bash know that it must look at the specified file
`source ~/.config/user-dirs.dirs`.

This value, which is created by `xdg-user-dirs-update` will vary depending on the language your system is set to. On my german system it says 

> XDG_TEMPLATES_DIR="$HOME/Vorlagen"

In my english system it says

> XDG_TEMPLATES_DIR="$HOME/Templates"

That translation is automatically being done by `xdg-user-dirs-update`. So basically what I do is to check whether the `config/user-dirs.dirs` file exists or not.
If it doesn't, I create it. If it does, I'll tell the terminal to look into `~/.config/user-dirs.dirs` and to take the `$XDG_TEMPLATES_DIR` value to assign it to the `customPath` string.

So I didn't set the value in the value of 

> $XDG_TEMPLATES_DIR

in the script, because it is automatically being created by the system.
That way, the user could also edit the file to 

> XDG_TEMPLATES_DIR="$HOME/MyFunnyCats"

which isn't a translation of "templates" in any language, but the user could use this directory if he/she wishes to have a directory named MyFunnyCats functioning as the Templates directory.

angela-d commented on automate.sh:
```diff
@@ -1,45 +1,45 @@
 
-		# this config already exists; append the entry
-		source ~/.config/user-dirs.dirs
-		if [[ $XDG_TEMPLATES_DIR = "$HOME/Templates" ]]
```
Do a check like this! :)

GithubUser699 replied:
The problem I have is that I don't know all the possible translated names and that some users might also change their assigned path name manually (see my other comment above).
Also, if a new language is being supported by Gnome, the script (my pull request script) would automatically take that associated name  and we wouldn't have to pull an update script.

I don't know if there is still a way to do such a check, but as of now I cannot think of any way. If you have an idea I'd be glad to hear it from you.

As far as I know this code

> if [[ $XDG_TEMPLATES_DIR = "$HOME/Templates" ]]

looks only whether there is an entry for `$XDG_TEMPLATES_DIR` (in that case if the value is `"$HOME/Templates"`) so we would have to find a method to check whether the entry is not null.

Did I get your concern right or did you mean something else?

GithubUser699 replied:
I just figured it out and it was really easy. Should have been obvious ;)

Here's the file and commit: https://github.com/GithubUser699/nautilus-right-click-new-file/commit/810a9923555845db68b0080d9bbbd915e4cabdef

GithubUser699 committed:
Update automate.sh

I just solved a bug where running the script without having the '~/.config/user-dirs.dirs' file will only execute 'xdg-user-dirs-update' without adding the right click menu files.
To solve this, I just re-execute the script automatically, so the user doesn't have to do this by their own.
